plugins/modules/vmware_vmkernel.py

```diff
@@ -801,20 +796,43 @@ def set_vsan_service_type(self):
         result = None
         vsan_system = self.esxi_host_obj.configManager.vsanSystem
 
-        vsan_port_config = vim.vsan.host.ConfigInfo.NetworkInfo.PortConfig()
-        vsan_port_config.device = self.vnic.device
-
+        vsan_system_config = vsan_system.config
         vsan_config = vim.vsan.host.ConfigInfo()
-        vsan_config.networkInfo = vim.vsan.host.ConfigInfo.NetworkInfo()
-        vsan_config.networkInfo.port = [vsan_port_config]
-        if not self.module.check_mode:
+
+        vsan_config.networkInfo = vsan_system_config.networkInfo
+        current_vsan_vnics = [portConfig.device for portConfig in vsan_system_config.networkInfo.port]
+        changed = False
+        result = "%s NIC %s (currently enabled NICs: %s) : " % ("Enable" if enable_vsan else "Disable", self.vnic.device, current_vsan_vnics)
+        if not enable_vsan:
+            if self.vnic.device in current_vsan_vnics:
+                vsan_config.networkInfo.port = list(filter(lambda portConfig: portConfig.device != self.vnic.device, vsan_config.networkInfo.port))
+                changed = True
+        else:
+            if self.vnic.device not in current_vsan_vnics:
+                vsan_port_config = vim.vsan.host.ConfigInfo.NetworkInfo.PortConfig()
+                vsan_port_config.device = self.vnic.device
+
+                if vsan_config.networkInfo is None:
+                    vsan_config.networkInfo = vim.vsan.host.ConfigInfo.NetworkInfo()
+                    vsan_config.networkInfo.port = [vsan_port_config]
+                else:
+                    vsan_config.networkInfo.port += [vsan_port_config]
```

Comment:
```suggestion
                    vsan_config.networkInfo.port.append(vsan_port_config)
```
I think this would be a bit more pythonic, but I'm not really sure. You don't have to change this, just a suggestion.

Reply:
I went for visual similarities, but you're probably right in that it's better to append an element directly as opposed to creating a list to be thrown away the next second.